src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Enumerated.cs

```diff
@@ -329,53 +286,31 @@ public ReadOnlyMemory<byte> ReadEnumeratedBytes(Asn1Tag expectedTag)
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method
         /// </exception>
-        public TEnum ReadEnumeratedValue<TEnum>(Asn1Tag expectedTag) where TEnum : struct
+        public TEnum ReadEnumeratedValue<TEnum>(Asn1Tag? expectedTag = null) where TEnum : Enum
         {
-            Type tEnum = typeof(TEnum);
+            TEnum ret =
+                AsnDecoder.ReadEnumeratedValue<TEnum>(_data.Span, RuleSet, out int consumed, expectedTag);
```

Comment:
Nit: the 9 additional characters it would add to line length was too much, hence the newline + indenting?

Reply:
Hm, I thought my VS horizontal scrollbar kicked in when the GitHub default width horizontal scrollbar kicked in.  Looks like mine fired early.

Aside from where the boundary condition is, the concept was a correct assessment, though :)